src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxOperation.cs

```diff
@@ -118,5 +127,32 @@ protected DacDeployOptions GetDefaultDeployOptions()
 
             return options;
         }
+
+        /// <summary>
+        /// Changes telemetry status 
+        /// </summary>
+        private void UpdateTelemetryStatus()
+        {
+            try
+            {
+                DacExtensions.SetServiceSetting(this.DacServices, "TelemetryEnabled", TelemetryEnabled);
+                DacExtensions.SetServiceSetting(this.DacServices, "TelemetryApplicationName", TelemetryApplicationName);
```

Comment:
You should only set the app name and version once on startup - no reason to reset it every time. Only the enabled state should change and even that only when the config changes.

Reply:
DacFx settings need to set for each operation. this is by design for DacFx

Reply:
Why did you make it per services instance anyways though? Telemetry should be enabled/disabled state I would think - is there a scenario where you specifically wanted to have it enabled for one operation but not for another or something?

Reply:
yes, user can do an import with telemetry then disable the telemetry and do export ...

Reply:
for each DacFx operation we create a new DacSerive and telemetry will be enabled at the service level

Reply:
Is there a specific scenario where you thought that would be useful? It just seems easier and more expected to have the telemetry toggle be at the global level. I can't think of any tools that have it more fine grained than that.

Reply:
Anyways it is what it is for now, but I would strongly suggest going back and thinking about whether you really want to keep it on a per-service level. It's not too late to change it IMO and would make it a lot easier and safer to ensure you're not sending events that you don't intend to.

Reply:
sure I can add a new API in DacFx to enable telemetry globally so user doesn't have to do it eveytime creating dac service. I think that make sense